_posts/2020-11-04-solidity-ama-1-recap.md

```diff
@@ -35,35 +35,80 @@ You can check the progress [here](https://github.com/ethereum/solidity/pull/1001
 
+There is single stage, the "constant optimizer" where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant. The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
+
 One of the big advantages of the Yul-based optimizer is that each step can be seen in isolation: Each step receives Yul code as input and produces Yul code as output, without any tight dependency on other steps or analysis code. Furthermore, we try to keep each step as simple as possible so that bugs in those steps are very unlikely. As long as each simple step is bug-free, so is the whole Yul optimizer.
 
 #### How does it work?
 
-There is single stage, the "constant optimizer" where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant. The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
-
 There are several steps that the optimizer performs. A simple example would be evaluating expressions whose value is known at runtime, e.g., `x + 0` is evaluated to `x`, where `x` may be a parameter known only during runtime. A more complex example would be to identify expressions that remains invariant inside a loop and moving them outside the loop, thereby saving gas. Another interesting example would be avoiding accessing the same value from storage multiple times, i.e., multiple `sload` to the same slot can be reduced to a single `sload`, in certain situations.
```

Comment:
I think this section is a good, high-level description of what optimizer does (both old and new one). I'd move it above "What is the Solidity Optimizer optimizing for".

Reply:
Looks like you missed this one. Or do you think that it's better to leave it where it is?

Reply:
if we move this block up the "how does it work" section is empty, so I'd keep it here. :)

Reply:
Yeah, but I meant the whole section. "how does it work" does sound like a more general question so it would make sense to have it higher to introduce the topic first.